api.go

```diff
@@ -64,6 +64,9 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 		return nil, err
 	}
 
+	// Unlock the thread here after the network hook was run.
+	runtime.UnlockOSThread()
```

Comment:
If we know that we don't need the lockOSThread after a certain point, we should call unlock() at that point, not in every single entry point?